packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkStateManager.ts

```diff
@@ -468,4 +428,19 @@ export class ForkStateManager implements EIP2929StateManager {
     this._stateRoot = newRoot;
     this._state = state;
   }
+
+  public async hasStateRoot(root: Buffer): Promise<boolean> {
+    return this._state.has(bufferToHex(root));
+  }
+
+  public async flush(): Promise<void> {
+    // not implemented
```

Comment:
We should probably throw

Reply:
These methods are called by the EEI, so we can't throw. No-op'ing is fine though, because they are just called (they don't return anything).